node/node.go

```diff
@@ -423,17 +423,21 @@ func (app *App) Initialize() (err error) {
 	var existing config.GenesisConfig
 	if err := existing.LoadFromFile(gpath); err != nil {
 		if !errors.Is(err, os.ErrNotExist) {
+			app.fileLock.Unlock()
```

Comment:
I don't think we can

Reply:
Why not? It would be called in case of panic if it was deferred.

Reply:
If I `defer app.fileLock.Unlock()` then the lock will be released when `Initialize` returns which is not what we want.

Reply:
Of course not, it would need some refactoring.

Reply:
good point, the current logic makes no sense (it locks too late and unlocks too early).

I created 2 new methods: `app.Lock()` and `app.Unlock`, the first of which is called earlier than `Initialize` and if successful is immediately followed by a `defer app.Unlock` so that unlocking is the last thing the node does (instead of the first thing when shutting down).